File: android/app/src/main/kotlin/com/runtrack/data/models/RunningRecord.kt
package com.runtrack.data.models

import kotlinx.serialization.Serializable

@Serializable
data class RunningRecord(
    // TODO: PM을 참고하여 러닝 기록 모델 단연 정의
)
    val difficulty: String? = null,
    val feeling: String? = null,
    val notes: String? = null,
    val createdAt: String,
    val updatedAt: String,
)

@Serializable
data class RunningRecordCreateRequest(
    val title: String? = null,
    val description: String? = null,
    val distanceKm: Float,
    val durationMinutes: Int,
    val startTime: String,
    val endTime: String,
    val shoeId: Int? = null,
    val avgPaceMinPerKm: Float? = null,
    val maxSpeedKmh: Float? = null,
    val avgSpeedKmh: Float? = null,
    val avgHeartRate: Int? = null,
    val caloriesBurned: Float? = null,
    val elevationGainM: Float? = null,
    val temperatureC: Float? = null,
    val feeling: String? = null,
    val notes: String? = null,
)

@Serializable
data class RunningStatistics(
    val totalDistanceKm: Float,
    val totalRuns: Int,
    val avgDistanceKm: Float,
    val avgPaceMinPerKm: Float,
    val totalDurationMinutes: Int,
    val avgDurationMinutes: Int,
    val totalCalories: Float,
    val avgHeartRate: Float? = null,
    val avgElevationGainM: Float? = null,
)

File: backend/app/schemas/running_record.py
"""
러닝 기록 Pydantic 스키마
"""
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class RunningRecordBase(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    distance_km: float
    duration_minutes: int
    start_time: datetime
    end_time: datetime

class RunningRecordCreate(RunningRecordBase):
    shoe_id: Optional[int] = None
    avg_pace_min_per_km: Optional[float] = None
    max_speed_kmh: Optional[float] = None
    avg_speed_kmh: Optional[float] = None
    avg_heart_rate: Optional[int] = None
    elevation_gain_m: Optional[float] = None
    temperature_c: Optional[float] = None
    feeling: Optional[str] = None
    notes: Optional[str] = None

class RunningRecordUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    shoe_id: Optional[int] = None
    feeling: Optional[str] = None
    notes: Optional[str] = None

class RunningRecordResponse(RunningRecordBase):
    id: int
    user_id: int
    shoe_id: Optional[int]
    avg_pace_min_per_km: Optional[float]
    max_speed_kmh: Optional[float]
    avg_speed_kmh: Optional[float]
    avg_heart_rate: Optional[int]
    calories_burned: Optional[float]
    elevation_gain_m: Optional[float]
    temperature_c: Optional[float]
    feeling: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class RunningStatistics(BaseModel):
    total_distance_km: float
    total_runs: int
    avg_distance_km: float
    avg_pace_min_per_km: float
    total_duration_minutes: int
    avg_duration_minutes: int
    total_calories: float
    avg_heart_rate: Optional[float]
    avg_elevation_gain_m: Optional[float]
